src/kernel/src/fs/null/mod.rs

```diff
@@ -0,0 +1,126 @@
+use super::{FsConfig, FsOps, LookupError, Mount, MountFlags, MountOpts, VPathBuf, Vnode};
+use crate::errno::{Errno, EDEADLK, EINVAL, EOPNOTSUPP};
+use crate::fs::VPath;
+use crate::ucred::Ucred;
+use bitflags::bitflags;
+use std::num::NonZeroI32;
+use std::sync::Arc;
+use thiserror::Error;
+
+mod vnode;
+
+pub fn mount(
+    _conf: &'static FsConfig,
+    _cred: &Arc<Ucred>,
+    _path: VPathBuf,
+    _parent: Option<Arc<Vnode>>,
+    mut opts: MountOpts,
+    flags: MountFlags,
+) -> Result<Mount, Box<dyn Errno>> {
+    if flags.intersects(MountFlags::MNT_ROOTFS) {
+        Err(MountError::RootFs)?;
+    }
+
+    if flags.intersects(MountFlags::MNT_UPDATE) {
+        if opts.remove("export").is_some_and(|opt| opt.unwrap()) {
+            todo!("null_mount with MNT_UPDATE and export = true")
+        } else {
+            Err(MountError::NoExport)?
+        }
+    }
+
+    let target: Box<str> = opts
+        .remove("target")
+        .or_else(|| opts.remove("from"))
+        .ok_or(MountError::NoTarget)?
+        .unwrap();
+
+    if target.is_empty() {
+        Err(MountError::EmptyTarget)?;
+    }
+
+    let _target: &VPath = target.as_ref().try_into().unwrap();
+
+    todo!()
+}
+
+fn root(_mnt: &Arc<Mount>) -> Arc<Vnode> {
+    todo!()
+}
+
+pub(super) static NULLFS_OPS: FsOps = FsOps { root };
+
+#[derive(Debug, Error)]
+#[allow(dead_code)]
```

Comment:
This one should be only applied to the PS4 types that we need to keep any discovered fields.